DevelopGitHub/GitHubExtension/GitHubExtension.Security.Tests/TestForControllers/TestGitHubExtensionSecurityWebApiAccountControllerAssignRolesToUser.cs

```diff
@@ -0,0 +1,181 @@
+using NSubstitute;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Data.Entity;
+using System.Linq;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+using System.Web.Http;
+using System.Web.Http.ModelBinding;
+using System.Web.Http.Results;
+using Xunit;
+
+namespace GitHubExtension.Security.Tests.TestForControllers
+{
+    public class TestGitHubExtensionSecurityWebApiAccountControllerAssignRolesToUser
+    {
+        [Fact]
+        public void CheckUserNotFound()
+        {
+            int gitHubId = 12;
+            int repoId = 0;
+            string roleToAssign = "role";
+            var store = Substitute.For<IUserStore<User>>();
+            var userManager = Substitute.For<ApplicationUserManager>(store);
+            List<User> users = new List<User>{
+                new User{ProviderId = 1,},
+                new User{ProviderId =2, },
+                new User{ProviderId=3,},
+            };
+
+            userManager.Users.Returns(new MockForEnumerableQuery<User>(users));
+
+            AccountController controller = new AccountController(Substitute.For<IGithubService>(),
+                Substitute.For<ISecurityContext>(), Substitute.For<IAuthService>(), userManager, store,
+                Substitute.For<IRoleStore<IdentityRole, string>>());
+
+            IHttpActionResult nullUser = controller.AssignRolesToUser(repoId, gitHubId, roleToAssign).Result;
+
+            Assert.IsType<NotFoundResult>(nullUser);
+        }
+
+        [Fact]
+        public void CheckStatusCodeIfRoleNotExists()
+        {
+            int gitHubId = 1;
+            int repoId = 0;
+            string roleToAssign = "BussinessAnalytics";
+            var store = Substitute.For<IUserStore<User>>();
+            var userManager = Substitute.For<ApplicationUserManager>(store);
+            List<User> users = new List<User>{
+                new User{ProviderId = 1,},
+                new User{ProviderId =2, },
+                new User{ProviderId=3,},
+            };
+
+            userManager.Users.Returns(new MockForEnumerableQuery<User>(users));
+            var context = Substitute.For<ISecurityContext>();
+            IEnumerable<SecurityRole> roles = new List<SecurityRole>
+            {
+                new SecurityRole{ Id = 1, Name = "Admin" },
+                new SecurityRole{ Id = 1, Name = "Developer" },
+                new SecurityRole{ Id = 1, Name = "Reviewer" }
+            };
+
+            context.SecurityRoles.Returns(new MockForDbSet<SecurityRole>(roles));
+
+            
+
+            AccountController controller = new AccountController(Substitute.For<IGithubService>(),
+                context, Substitute.For<IAuthService>(), userManager, store,
+                Substitute.For<IRoleStore<IdentityRole, string>>());
+
+            IHttpActionResult nullUser = controller.AssignRolesToUser(repoId, gitHubId, roleToAssign).Result;
```

Comment:
Function call and getting result should be in different unit test phases, respectively Act and Assert.